lib/utils/change-object.js

```diff
@@ -1,15 +1,34 @@
 var assign = require('lodash/assign')
+var validate = require('../validate')
 
 /**
   * change object either by passing changed properties
   * as an object, or by passing a change function that
   * manipulates the passed object directly
   **/
-module.exports = function changeObject (change, object) {
+module.exports = function changeObject (state, change, object) {
+  var updatedObject = assign({}, object)
+
   if (typeof change === 'object') {
-    return assign(object, change)
+    updatedObject = assign(object, change)
+  } else {
+    change(updatedObject)
   }
 
-  change(object)
-  return object
+  if (state === null) {
```

Comment:
This feels wrong to me. I can’t do a thorough review right now I’m afraid, I’ll be at conferences the next two weeks and barely online. Just from looking at the code, I’m not sure when `state` can be null? It feels like an anti pattern. Either you pass in state or not

Reply:
Yes, even I have concerns over the same thing. It feels super hacky. I need some way to figure out if validation is required or not. 
Should I add a `needValidation` argument to the function definition, instead of passing in `null` for the `state` placeholder in the function?

Reply:
I don’t know, I’ll need more time to look into it, unfortunately I’ll be gone for two weeks, just want to let you know. Feel free to experiment though, maybe you come up with a clear way :)